src/fsharp/FSharp.Core/string.fsi

```diff
@@ -98,26 +122,115 @@ namespace Microsoft.FSharp.Core
         ///
         /// <returns>The constructed string.</returns>
         /// <exception cref="T:System.ArgumentException">Thrown when <c>count</c> is negative.</exception>
+        /// 
+        /// <example> Enumerate digits ASCII codes
+        /// <code lang="fsharp">
+        /// String.init 10 (fun i -> int '0' + i |> sprintf "%d ")  // evaluates "48 49 50 51 52 53 54 55 56 57 "
+        /// </code>
+        /// </example>
         [<CompiledName("Initialize")>]
         val init: count:int -> initializer:(int -> string) -> string
 
-        /// <summary>Tests if all characters in the string satisfy the given predicate.</summary>
+        /// <summary>Applies the function <c>action</c> to each character in the string.</summary>
         ///
-        /// <param name="predicate">The function to test each character of the string.</param>
+        /// <param name="action">The function to be applied to each character of the string.</param>
         /// <param name="str">The input string.</param>
+        /// 
+        /// <example>
+        /// <code lang="fsharp">
+        /// String.iter (fun c -> printfn "%c %d" c (int c)) "Hello"  
+        /// // evaluates unit
+        /// // prints:
+        /// H 72
+        /// e 101
+        /// l 108
+        /// l 108
+        /// o 111
+        /// </code>
+        /// </example>
+        [<CompiledName("Iterate")>]
+        val iter: action:(char -> unit) -> str:string -> unit
+
+        /// <summary>Applies the function <c>action</c> to the index of each character in the string and the
+        /// character itself.</summary>
         ///
-        /// <returns>True if all characters return true for the predicate and false otherwise.</returns>
-        [<CompiledName("ForAll")>]
-        val forall: predicate:(char -> bool) -> str:string -> bool
+        /// <param name="action">The function to apply to each character and index of the string.</param>
+        /// <param name="str">The input string.</param>
+        /// 
+        /// <example>
+        /// <code lang="fsharp">
+        /// String.length null   // evaluates 0
+        /// String.length ""     // evaluates 0
+        /// String.length "123"  // evaluates 3
+        /// </code>
+        /// </example>
+        /// 
+        /// <example>
```

Comment:
This is good.  Separating into multiple examples is totally fine. Do the same for those further above if you think it's right